inc/knowbaseitem.class.php

```diff
@@ -1262,61 +1309,67 @@ static function getListRequest(array $params, $type = 'search') {
                                    /* 8 */   "/\)/",
                                    /* 9 */   "/\-/"];
                   $contains = preg_replace($search1, "", $params["contains"]);
-                  $addwhere = '';
+                  $ors = [
+                     "glpi_knowbaseitems.name ".Search::makeTextSearch($contains),
+                     "glpi_knowbaseitems.answer ".Search::makeTextSearch($contains)
+                  ];
                   if (KnowbaseItemTranslation::isKbTranslationActive()
                       && (countElementsInTable('glpi_knowbaseitemtranslations') > 0)) {
-                     $addwhere = " OR `glpi_knowbaseitemtranslations`.`name` ".Search::makeTextSearch($contains)."
-                                    OR `glpi_knowbaseitemtranslations`.`answer` ".Search::makeTextSearch($contains);
+                     $ors[] = "glpi_knowbaseitemtranslations.name ".Search::makeTextSearch($contains);
+                     $ors[] = "glpi_knowbaseitemtranslations.answer ".Search::makeTextSearch($contains);
                   }
-                  $where   .= " AND (`glpi_knowbaseitems`.`name` ".Search::makeTextSearch($contains)."
-                                 OR `glpi_knowbaseitems`.`answer` ".Search::makeTextSearch($contains)."
-                                 $addwhere)";
+                  $criteria['WHERE'][] = ['OR' => $ors];
                } else {
-                  $where = $where_1;
+                  $criteria['WHERE'] = $search_where;
+                  // Add visibility date
+                  $criteria['WHERE'][] = [
```

Comment:
This was previously after closing brace of if/else block.

Reply:
Same condition were added twice then as far as I've understand

Reply:
In fact, here, as you build your search from `$search_where`, you already have these criteria. But in the `if` block just above, you do not have them. Moving addition of these criteria in the `else` block instead of outside `if/else` will change the behaviour.